### PR TITLE
[DLP] Implemented dlp_deidentify_table_row_suppress with unit test cases

### PR DESCRIPTION
**Pull request by Avani-Thakker-Crest**
## Description
[DLP] Implemented dlp_deidentify_table_row_suppress with unit test cases. Java equivalent: https://cloud.google.com/dlp/docs/examples-deid-tables.md#dlp-deid-table-row-suppress-java

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [x] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] Please **merge** this PR for me once it is approved.
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample

### CODE REVIEW

File: dlp/snippets/deid.py [hunk 1]
```diff
@@ -1282,6 +1399,36 @@ def deidentify_with_exception_list(
...
+        "table_data",
+        help="Json string representing table data",
+    )
+    table_row_suppress_parser.add_argument(
+        "--condition_field",
+        help="A table Field within the record this condition is evaluated "
+             "against.",
+    )
+    table_row_suppress_parser.add_argument(
+        "--condition_operator",
```
Comment (annadata) [inline dlp/snippets/deid.py:1421]: Some of the arguments are present in another PR as well? Needs resolving conflicts.

Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid.py:1421]: Arguments are for different functions.  there would be no conflict.

File: dlp/snippets/deid_test.py [hunk 2]
```diff
@@ -305,3 +313,18 @@ def test_deidentify_with_exception_list(capsys):
...
+        GCLOUD_PROJECT,
+        TABLE_DATA,
+        "age",
+        "GREATER_THAN",
+        89
+    )
+
+    out, _ = capsys.readouterr()
+
+    assert "string_value: \"Jane Austen\"" in out
```
[the remark this answers is not in the document]
Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid_test.py]: Thank you for reviewing. Added assertion for Mark Twain for completeness. (Mark twain will be suppressed as his age is above 89)

File: dlp/snippets/deid.py [hunk 3]
```diff
@@ -1023,6 +1023,123 @@ def deidentify_with_exception_list(
...
+        table_data: Json string representing table data.
+        condition_field: A table Field within the record this condition is evaluated against.
+        condition_operator: Operator used to compare the field or infoType to the value. One of:
+            RELATIONAL_OPERATOR_UNSPECIFIED, EQUAL_TO, NOT_EQUAL_TO, GREATER_THAN, LESS_THAN, GREATER_THAN_OR_EQUALS,
+            LESS_THAN_OR_EQUALS, EXISTS.
+        condition_value: Value to compare against. [Mandatory, except for ``EXISTS`` tests.].
+
+    Returns:
+        De-identified table is returned;
+        the response from the API is also printed to the terminal.
```
Comment (realjordanna) [inline dlp/snippets/deid.py:1048]: Capitalize "the"

**Review by realjordanna — CHANGES_REQUESTED** [review]

File: dlp/snippets/deid.py [hunk 4]
```diff
@@ -1023,6 +1023,123 @@ def deidentify_with_exception_list(
...
+
+    # Import the client library
+    import google.cloud.dlp
+
+    # Instantiate a client.
+    dlp = google.cloud.dlp_v2.DlpServiceClient()
+
+    # Construct the `table`. For more details on the table schema, please see
+    # https://cloud.google.com/dlp/docs/reference/rest/v2/ContentItem#Table
+    headers = [{"name": val} for val in table_data["header"]]
```
[the remark this answers is not in the document]
Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid.py:1088]: Hi @realjordanna thank you for your comment. I am not very sure how to do this, I tried looking into it, I found we should be importing pb2 files from google.privacy.dlp.v2 package, but I failed to do so. I have worked with protocol buffers in the past and to use those files we must compile .proto files and that generates pb2 files.. I feel I am missing some steps here to import pb2 files from google.privacy.dlp.v2 package. Please guide me through; that would be helpful, thank you

[the remark this answers is not in the document]
Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid.py:1088]: I am receiving compiler error while importing google.privacy.dlp.v2 package.

[the remark this answers is not in the document]
Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid.py:1624]: I will try to revisit this example using PB later. For now based on discussion with @annadata  we will use the current approach. Thank you! Closing the comment for now.

File: dlp/snippets/deid.py (hunk 3 above)
Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid.py]: added type hints, removed return table as mentioned in one of the PRs, "Returns" in the doc string is removed as it returns none now

**Review by kweinmeister — APPROVED** [review]

**Review by realjordanna — APPROVED** [review]

**Review by realjordanna — APPROVED** [review]

File: dlp/snippets/deid.py [hunk 5]
```diff
@@ -1125,15 +1125,17 @@ def deidentify_with_exception_list(
 
 
 # [START dlp_deidentify_table_bucketing]
+from typing import Dict  # noqa: F811, E402, I100
 from typing import List  # noqa: F811, E402, I100
+from typing import Union  # noqa: F811, E402, I100
```
Comment (rsamborski) [inline dlp/snippets/deid.py:1130]: nit: consider putting this in a single import, i.e. `from typing import Dict, List, Union`

File: dlp/snippets/deid.py [hunk 6]
```diff
@@ -1246,20 +1228,22 @@ def deidentify_table_bucketing(
 
 
 # [START dlp_deidentify_table_condition_infotypes]
+from typing import Dict  # noqa: F811, E402, I100
 from typing import List  # noqa: F811, E402, I100
+from typing import Union  # noqa: F811, E402, I100
```
Comment (rsamborski) [inline dlp/snippets/deid.py:1233]: ditto

File: dlp/snippets/deid.py [hunk 7]
```diff
@@ -1386,19 +1350,21 @@ def deidentify_table_condition_replace_with_info_types(
 
 
 # [START dlp_deidentify_table_condition_masking]
+from typing import Dict  # noqa: F811, E402, I100
 from typing import List  # noqa: F811, E402, I100
+from typing import Union  # noqa: F811, E402, I100
```
Comment (rsamborski) [inline dlp/snippets/deid.py:1355]: ditto

File: dlp/snippets/deid.py [hunk 8]
```diff
@@ -1619,6 +1567,101 @@ def deidentify_table_replace_with_info_types(
 # [END dlp_deidentify_table_infotypes]
 
 
+# [START dlp_deidentify_table_row_suppress]
+from typing import Dict  # noqa: F811, E402, I100
+from typing import List  # noqa: F811, E402, I100
+from typing import Union  # noqa: F811, E402, I100
```
Comment (rsamborski) [inline dlp/snippets/deid.py:1573]: ditto

**Review by rsamborski — APPROVED** [review]
Approved with minor suggestion

File: dlp/snippets/deid.py (hunk 5 above)
Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid.py]: updated

**Comment by engelke** [comment]
Removing conflicts when merging from main seems to have lead to a lost definition. Asking the author to address what I may have messed up. Sorry!

**Comment by Avani-Thakker-Crest** [comment]
> 

No worries, I have resolved the conflicts, have rebased with main and it seems fine now. Please check and Thank you for your efforts :)

**Pull request merged**